docs/4-functional-programming/fp-hometask/src/utils.ts

```diff
@@ -0,0 +1,58 @@
+import { Row } from './components';
+import { FilterValueType, SortValueType } from './types';
+
+export function filterBySearch(rows: Row[], value: string): Row[] {
+  if (!rows || !value) {
+    return [];
+  }
+
+  return [...rows].filter(
+    ({ username, country, name }) =>
+      username.toLowerCase().includes(value.toLowerCase()) ||
+      country.toLowerCase().includes(value.toLowerCase()) ||
+      name.toLowerCase().includes(value.toLowerCase())
+  );
+}
+
+export function filterByFilters(rows: Row[], value: string[]): Row[] {
+  if (!rows || !value.length) {
+    return [];
+  }
+
+  if (value.length === 2) {
+    return rows.filter(({ posts }) => !posts || posts > 100);
+  }
+
+  if (value.includes(FilterValueType.WithoutPosts)) {
+    return rows.filter(({ posts }) => !posts);
+  }
+
+  if (value.includes(FilterValueType.MoreThan100Posts)) {
+    return rows.filter(({ posts }) => posts > 100);
+  }
+}
+
+export function sortRows(rows: Row[], sortValue: string) {
```

Comment:
This function mutates rows side effects.
Please, update the function to use in FP-like way

Reply:
done